routers/web/user/profile.go

```diff
@@ -108,74 +49,56 @@ func Profile(ctx *context.Context) {
 	}
 
 	// check view permissions
-	if !models.IsUserVisibleToViewer(ctxUser, ctx.User) {
-		ctx.NotFound("user", fmt.Errorf(uname))
-		return
-	}
-
-	// Show SSH keys.
-	if isShowKeys {
-		ShowSSHKeys(ctx, ctxUser.ID)
-		return
-	}
-
-	// Show GPG keys.
-	if isShowGPG {
-		ShowGPGKeys(ctx, ctxUser.ID)
-		return
-	}
```

Comment:
Is this just removed or is it moved elsewhere? I can't find it but maybe I missed it. If it is removed, could please explain why?

Reply:
Note to other reviewers: there was a hint in the description.

> While most changes are straight forward, have a look at [routers/web/user/profile.go](https://github.com/go-gitea/gitea/compare/main...KN4CK3R:refactor-context-user?expand=1#diff-4b6486152ab1c16e13e39847e4b7a848f2981d55a1b2390c36cd1bec167e545f) where I had to change the Profile method with additional routes.